var/spack/repos/builtin/packages/pmix/package.py

```diff
@@ -101,6 +114,14 @@ def configure_args(self):
         config_args.append("--with-libevent=" + spec["libevent"].prefix)
         config_args.append("--with-hwloc=" + spec["hwloc"].prefix)
 
+        if "64" in platform.machine():
```

Comment:
I'm mostly happy with this PR, but is this check correct? Does it work on 32-bit?

Reply:
I tried installing the hwloc version 2.1 available with ubuntu20.04 (i386 version). Then I tried building pmix with spack without the 64-bit check. Pmix build failed in the configuration stage itself as it incorrectly deemed the version of hwloc < 1.5. So, I am not even sure if pmix build needs to be fixed before trying to add stuff to spack. Since, I've tested this only on a 64-bit platform and it works with 64-bit library binaries, that's why I have left this check.

I wonder if we can work on the 32-bit support when needed and have this 64-bit check in place at the moment. If that seems reasonable then I can leave a comment explaining how to use find_external_lib_path to accomplish 32-bit support.